### ISSUE TITLE
Train model on custom uint16 single channel xray images

### ISSUE THREAD

**Issue by fabiozappo** [ultralytics/yolov5#1342]
Hi Glenn,

Thank you for sharing your model with such a good documentation and easy readable code, can't wait to read the incoming paper! I am here cause I think it's a huge improvement for object detection expecially for custom tasks such mine. I work with dicom files (used for medical analysis) containing both x-ray 8k x 6k grayscale uint16 images and annotations with two classes and very small bounding boxes (between 10 and 50 pixels each one). Information in this image is very important and I can not resize image dimensions and downcast to 8bit images.
I easily found the code parts to modify in order to load image at uint16 and preprocess images with the right normalization (img/255 --> img/65535). I still work with three channels in order to finetune your pretrained coco weights.
Results are quite good (60% map), however i would like to improve them in order to reach a higher map requested for my research.
 
I would appreciate if you confirm my doubts:

1) In order to work with such huge images my idea is to slice each image in several 1024x1024 crops and then train the model directly on them. Same method will be used for inference process. Is it a good idea?
2) Is it a good idea to train from scratch the model on coco grayscale images in order to finetune the final model with single channel? Or do you suggest to keep using 3 channels image?

If you have any other suggestions for my dataset i would really appreciate and it would be also a good thing for medical x-ray images object detection.

Thank you for your time.

**Comment by glenn-jocher** [comment]
@fabiozappo for large images such as xrays and satellite imagery you can use a sliding window dataloader to extract smaller tiles at native resolution, though this requires additional complication and overhead for tile overlap, stitching etc. We used this on NGA satellite imagery here:
https://github.com/ultralytics/xview-yolov3

The simpler and best approach would be to simply train and test and native resolution or the minimum image size your hardware will allow at batch size 1, with my obvious response to CUDA OOM being that you need more/better hardware.

The input channel count is largely irrelevant, as it makes up an infinitesimally small portion of the model FLOPS. You can reconstruct the model with any input channel count you want, see PyTorch Hub tutorial in https://docs.ultralytics.com/yolov5, however I doubt this will have a material impact on your training results.

Good luck!

**Comment by glenn-jocher** [comment]
Also for very small batch sizes on multigpu you may want to try with and without --sync-bn.

**Comment by fabiozappo** [comment]
> for large images such as xrays and satellite imagery you can use a sliding window dataloader to extract smaller tiles at native resolution, though this requires additional complication and overhead for tile overlap, stitching etc. We used this on NGA satellite imagery here:
> https://github.com/ultralytics/xview-yolov3

This seems much more faster and accurate then simply cropping images, thank you!


> The input channel count is largely irrelevant, as it makes up an infinitesimally small portion of the model FLOPS. You can reconstruct the model with any input channel count you want, see PyTorch Hub tutorial in https://docs.ultralytics.com/yolov5, however I doubt this will have a material impact on your training results.

If I've correctly understood how cnn works I agree that changing the number of channel will reduce model memory and flops just for the network part between the input layer and the first convolutional layer, so the model will almost have the same inference time. However my doubt is still about the possible Map benefits in using grayscale images for pretraining. Pretraining the model on single channel image will help it to learn more usefull features for any custom grayscale dataset, won't it?

**Comment by jlgn123** [comment]
Hello Fabio, I am also interested in training on 16-bit colour depth images, would you mind sharing your approach?  Assume you modified the `cv2.imread()` call in `dataloaders.py`?  Any other modifications?  And how did it work out in the end?

**Comment by fabiozappo** [comment]
Hi, in my work I modified some code about data augmentation in dataset.py in order to no more clip at 255 pixel value, I also changed normalization (from 255 to 65535) on input tensor before feed forward step and other stuff for grayscale images.

However I suggest you to convert your image from 16 to 8 bit as a first step and then just use train.py. This will give you an idea on how this naive approach works.

**Comment by taruntiwarihp** [comment]
> Hi, in my work I modified some code about data augmentation in dataset.py in order to no more clip at 255 pixel value, I also changed normalization (from 255 to 65535) on input tensor before feed forward step and other stuff for grayscale images.
> 
> However I suggest you to convert your image from 16 to 8 bit as a first step and then just use train.py. This will give you an idea on how this naive approach works.

@fabiozappo - I'm also working on x-ray data 256px images and using yolov5 to training model, but I'm getting 50%map, can you please help me and guide me to increase accuracy .

**Comment by fabiozappo** [comment]
Hi @taruntiwarihp, as a first step I suggest you to give a look at the official [ Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

However, in our work main contributors in increasing the model accuracy were:

- a custom data augumentation pipeline (e.g. we didn't use histogram equalization and clahe as they modified the intrinsic aspect of defect in xray images). I suggest you to give a look at training image to understand what is harmful for your images.

 - we generated a lot of synthethic images simulating the presence of defects. Then we train models from scratch on those images and used results weights as the starting weights in the official training.

**Comment by taruntiwarihp** [comment]
Thank you for your reply

**Comment by chethanlreddy** [comment]
hi @fabiozappo can u give me link for model that is trained on  x-ray image so I can use it as weights for my model please help me with it.

**Comment by glenn-jocher** [comment]
Hi there! Unfortunately, we don't provide specific pretrained models on x-ray images. However, you can train your own model using a relevant dataset and then use those weights for further training or inference. If you need guidance on how to train your model, the [YOLOv5 documentation](https://docs.ultralytics.com/yolov5/) is a great resource. Good luck with your project! 😊
